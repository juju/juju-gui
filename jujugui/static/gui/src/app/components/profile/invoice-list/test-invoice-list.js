/* Copyright (C) 2018 Canonical Ltd. */

'use strict';
const React = require('react');
const shapeup = require('shapeup');
const enzyme = require('enzyme');

const ProfileInvoiceList = require('./invoice-list');

describe('Invoice Bundle List', function() {
  const rawInvoiceData = `[{
  	"number": "100001",
  	"status": "Paid",
  	"chargedTo": "card ending 1234",
  	"date": "12/05/2017"
  }, {
  	"number": "100002",
  	"status": "Declined: will retry",
  	"chargedTo": "card ending 1231",
  	"date": "12/05/2017"
  }]`;

  let invoices;
  let acl;

  beforeEach(() => {
    acl = shapeup.deepFreeze({isReadOnly: () => false});
    invoices = JSON.parse(rawInvoiceData);
  });

  const renderComponent = (options = {}) =>
    enzyme.shallow(
      <ProfileInvoiceList
        acl={options.acl || acl}
        baseURL="/gui/"
        data={options.invoices || invoices}
        user="lazypower@external"
      />
    );

  it('can render', () => {
    const wrapper = renderComponent({gisf: true});
    expect(wrapper).toMatchSnapshot();
  });

  it('can render without any invoices', () => {
    const wrapper = renderComponent({
      invoices: []
    });
    assert.deepEqual(wrapper.find('BasicTable').prop('rows'), []);
  });
});
