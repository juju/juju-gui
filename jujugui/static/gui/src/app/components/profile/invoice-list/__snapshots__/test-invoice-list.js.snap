// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoice Bundle List can render 1`] = `
<div
  className="profile-invoice-list"
>
  <div>
    <div
      className="v1"
    >
      <h2
        className="profile__title"
      >
        Payment history
      </h2>
    </div>
    <BasicTable
      headerClasses={
        Array [
          "profile__entity-table-header-row",
        ]
      }
      headerColumnClasses={
        Array [
          "profile__entity-table-header-column",
        ]
      }
      headers={
        Array [
          Object {
            "columnSize": 3,
            "content": "Status",
          },
          Object {
            "columnSize": 3,
            "content": "Invoice Number",
          },
          Object {
            "columnSize": 3,
            "content": "Charged to",
          },
          Object {
            "columnSize": 3,
            "content": "Date",
          },
        ]
      }
      rowClasses={
        Array [
          "profile__entity-table-row",
        ]
      }
      rowColumnClasses={
        Array [
          "profile__entity-table-column",
        ]
      }
      rows={
        Array [
          Object {
            "columns": Array [
              Object {
                "columnSize": 3,
                "content": "Paid",
              },
              Object {
                "columnSize": 3,
                "content": <a
                  href="#invoices/42"
                >
                  100001
                </a>,
              },
              Object {
                "columnSize": 3,
                "content": "card ending 1234",
              },
              Object {
                "columnSize": 3,
                "content": "12/05/2017",
              },
            ],
            "key": "100001",
          },
          Object {
            "columns": Array [
              Object {
                "columnSize": 3,
                "content": "Declined: will retry",
              },
              Object {
                "columnSize": 3,
                "content": <a
                  href="#invoices/42"
                >
                  100002
                </a>,
              },
              Object {
                "columnSize": 3,
                "content": "card ending 1231",
              },
              Object {
                "columnSize": 3,
                "content": "12/05/2017",
              },
            ],
            "key": "100002",
          },
        ]
      }
    />
  </div>
</div>
`;
